### PR TITLE
Feat: Adding logging features plus other updates

### PR DESCRIPTION
**Pull request by kipropbrian**
### PR Description
Adding debugging and logging if DEBUG=true in config plus other updates

### Description of tasks that were expected to be completed
- Added debugging functionality.
You need to edit .env. 

- Package.json
added `tsc --noEmit` to dev and seed so that typescript compiles and checks all files for type errors and fails if there are any. In tsconfig.json I ensured to ignore tests (*.spec.ts) and (*.d.ts) files.

- ES5 
Also removed es5 mixing of es5 and es6 similar to #193  and slightly refactored the code. 
 

### Functionality
Just set DEBUG=true in .env and you will be flooded with nice juicy fat logs.

### How has this been tested?
Got to .env
Set DEBUG=true

### PR Checklist:
N/A

### Track PR
N/A

### Screenshots (If appropriate)
  
Screenshot 2024-05-14 at 14 28 35

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,142 +1,133 @@
...
-      console.log(
-        `🚀 Server ready at http://localhost:${PORT}${server.graphqlPath}`
-      )
-    )
-  })
-})(typeDefs, resolvers)
+    console.log('Database Connected.');
+    console.log(`Environment is set to ${process.env.NODE_ENV}`);
+    httpServer.listen({ port: PORT }, () => {
+      console.log(`🚀 Server ready at http://localhost:${PORT}${server.graphqlPath}`);
```
Comment (ceelogre) [inline src/index.ts:128]: Did you intentionally leave these as `console.log`?

Comment (kipropbrian) [reply src/index.ts:128]: Yes, if I use the logger and set debug=false, this wont show. So I was thinking I leave this to display regardless of the DEBUG FLAG. Or maybe we should just put everything in the logger. What do you think is the better approach.

**Comment by kipropbrian** [comment]
By the way, what was this for. Should I just remove it. 

Screenshot 2024-05-15 at 11 21 36

File: src/index.ts (hunk 1 above)
Comment (ceelogre) [reply src/index.ts:128]: It's probably okay that way. There could be cases where we want to toggle the logger to off and we'd still want to see these.

**Pull request merged**